packages/stores/src/queries/concentrated-liquidity/account-positions.ts

```diff
@@ -58,17 +60,62 @@ export class ObservableQueryAccountPositions extends ObservableChainQuery<{
     );
   }
 
+  /** List of CL positions for account. */
   @computed
   get positions(): ObservableQueryLiquidityPositionById[] {
     return this.positionIds.map((id) => {
       return this.queryPositionsById.getForPositionId(id);
     });
   }
 
+  /** Aggregate of all coins in a user's list of CL positions. */
+  @computed
+  get totalPositionsAssets() {
+    return this.getPositionsAssets(this.positions);
+  }
+
+  /** Aggregated list of coins for a user's pool's positions. */
+  readonly totalPositionsAssetsInPool = computedFn((poolId: string) => {
+    return this.getPositionsAssets(this.positionsInPool(poolId));
+  });
+
   /** User account positions in a given pool of ID. */
   readonly positionsInPool = computedFn((poolId: string) => {
     return this.positions.filter((position) => position.poolId === poolId);
   });
+
+  /** Aggregates the coins in the given positions array. */
+  protected readonly getPositionsAssets = computedFn(
+    (positions: ObservableQueryLiquidityPositionById[]) => {
+      return Array.from(
+        positions
+          .reduce((balances, position) => {
+            const addToMap = (coin: CoinPretty) => {
```

Comment:
I think you might be able to simplify to avoid `if` `else`

```
const addToMap = (coin: CoinPretty) => {
  balances.set(
    coin.currency.coinMinimalDenom,
    (balances.get(coin.currency.coinMinimalDenom) || new CoinPretty(0)).add(coin)
  );
};

```